internal/providers/common/provider.go

```diff
@@ -1,10 +1,12 @@
-package provider
+// Package common contains the changes that depends on both internal/providers/sdkv2 and internal/providers/pluginfw packages
+package common
```

Comment:
I think the name "common" is more appropriate for shared dependencies (e.g. what is currently in `internal/provider.go`). What about putting this in `providers/tf6/tf6.go` or  even `providers/providers.go`?

I like `providers/providers.go` as it reinforces that this is wrapper of more specialized implementations.

Reply:
Moved them to providers.go to top level